src/Products/Annotation/Controllers/AnnotationApiController.cs

```diff
@@ -509,77 +393,80 @@ public HttpResponseMessage Annotate(AnnotationPostedDataEntity annotateDocumentR
                 // initiate list of annotations to add
-                List<AnnotationInfo> annotations = new List<AnnotationInfo>();
-              
-                DocumentInfoContainer documentInfo = AnnotationImageHandler.GetDocumentInfo(documentGuid, password);
-                // check if document type is image
-                if (SupportedImageFormats.Contains(Path.GetExtension(documentGuid)))
-                {
-                    documentType = "image";
-                }
-                // initiate annotator object  
-                string notSupportedMessage = "";
-                for (int i = 0; i < annotationsData.Length; i++)
+                List<AnnotationBase> annotations = new List<AnnotationBase>();
+
+                using (FileStream outputStream = File.OpenRead(documentGuid))
                 {
-                    // create annotator
-                    AnnotationDataEntity annotationData = annotationsData[i];
-                    PageData pageData = documentInfo.Pages[annotationData.pageNumber - 1];
-                    // add annotation, if current annotation type isn't supported by the current document type it will be ignored
-                    try
+                    using (GroupDocs.Annotation.Annotator annotator = new GroupDocs.Annotation.Annotator(outputStream, GetLoadOptions(password)))
                     {
-                        BaseAnnotator annotator = AnnotatorFactory.createAnnotator(annotationData, pageData);
-                        if (annotator.IsSupported(documentType))
-                        {
-                            annotations.Add(annotator.GetAnnotationInfo(documentType));
-                        }
-                        else
+                        string notSupportedMessage = "";
+                        IDocumentInfo info = annotator.Document.GetDocumentInfo();
+
+                        for (int i = 0; i < annotationsData.Length; i++)
                         {
-                            notSupportedMessage = annotator.Message;
+                            AnnotationDataEntity annotationData = annotationsData[i];
+                            PageInfo pageInfo = info.PagesInfo[annotationsData[i].pageNumber - 1];
+                            // add annotation, if current annotation type isn't supported by the current document type it will be ignored
+                            try
+                            {
+                                BaseAnnotator baseAnnotator = AnnotatorFactory.createAnnotator(annotationData, pageInfo);
+                                if (baseAnnotator.IsSupported(documentType))
+                                {
+                                    annotations.Add(baseAnnotator.GetAnnotationBase(documentType));
+                                }
+                                else
+                                {
+                                    notSupportedMessage = baseAnnotator.Message;
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this useless assignment to local variable 'notSupportedMessage'.](https://app.codacy.com/manual/GroupDocs/GroupDocs.Annotation-for-.NET-MVC/pullRequest?prid=5372470)